saleor/checkout/utils.py

```diff
@@ -1070,3 +1070,30 @@ def get_address_for_checkout_taxes(
 ) -> Optional["Address"]:
     shipping_address = checkout_info.delivery_method_info.shipping_address
     return shipping_address or checkout_info.billing_address
+
+
+def save_checkout_with_update_fields(checkout, update_fields: list[str]) -> None:
+    """Try to save the checkout object with providing fields to update.
+
+    The method is used to handle the DatabaseError that can occur when the save with
+    `update_fields` is performed on a deleted object. In such a case, the method raises
+    a ValidationError with the proper error.
+    """
+    removed_error = "Save with update_fields did not affect any rows."
```

Comment:
Should we add a unit test for base `checkout.save` that returns an error with this message? This should inform us if Django changes this message in any version.

Reply:
IMO I like the idea of separate test only for that on `save` method, as it will be less confusing if such scenario happens in the future.